### ISSUE TITLE
Feature request: set root on a selected folder, showing only its contents and its ancestors chain

### ISSUE THREAD

**Issue by chr7** [aaFn/Bookmark-search-plus-2#54]
I'm wondering if you could add the feature to have to bookmarks displayed  as a flat list (like [this](https://addons.mozilla.org/en-US/firefox/addon/flat-bookmarks-history/) instead of a tree.

Or if you would consider writing a separate plugin for that.

**Comment by aaFn** [comment]
Hello @chr7, BSP2 incorporates an horizontal scrollbar, which allows to see any depth of indentation, exactly for handling the main problem mentioned by the author of that old extension.
Since one of BSP2 purposes is to retrieve parent folder(s) of a bookmark, turning to a flat list would be a conflicting feature, destroying that purpose and BSP2 nature.

I believe it would fit better indeed as a separate add-on, whose purpose would be different (probably not working well by the way for people having lots of bookmarks ...) .. but I am not sure what would be the exact objective for that add-on. What do you think ?

**Comment by chr7** [comment]
I have more than 6000 bookmarks and a nested folder structure. I use the flat bookmark plugin with an older version of Firefox and I'm very happy with it. Unfortunately, it's not available for the latest version of Firefox thus my question.

A horizontal scroll bar is uncomfortable: the deeper the folder structure gets the more the bookmarks are indented thus you have a lot of space lost left of the bookmarks while the bookmarks get more and more less readable (not enough text displayed to the right - sure, I could increase the size of the bookmark pane but then I loose space for the main window). The space is used better with a flat bookmark structure.

I don't see any conflict with the feature to retrieve the parent folder (I use the Go Parent Folder extension on the older Firefox and it works perfect with the Flat Bookmark extension): just open it in the flat bookmark structure and show its content as a flat list of bookmarks.

**Comment by aaFn** [comment]
So you mean that the "flat list" still keeps the tree structure, so remains a tree view, but with no indentation, right ?
I.e bookmarks are still folded into folders, not visible as long as the folder is not open, and you view them only when the folder is open ?
(sorry, but never used that old extension and its description is quite cryptic ..)

**Comment by chr7** [comment]
Yes, the folder hierarchy is kept but without indentation and showing the content (bookmark and sub-folders) of the selected folder only.

![flatbookmarks](https://user-images.githubusercontent.com/7242780/44171606-e9c33800-a0da-11e8-84c2-98599f62d5ad.png)

**Comment by aaFn** [comment]
Ok, I understand :-)
So if you don't mind, I am changing the title of the issue to "Feature request: display bookmarks with no or little indentation".

Btw, on your picture, it looks likt there is a very small indentation of 1 or so pixel .. is that my eyes or am I right ?

Then it is easy to do .. I can add an option saying "no indentation" or "very small indentation (1 pixel)". And with the "Go parent" action on right click I added in 2.0.34, then you should have what you want.

Let me know.

**Comment by chr7** [comment]
I've never noticed it but indeed the folder content has a very small indentation - I'm fine with 1 pixel.

Please note that there is a bigger difference than just removing the indentation: the screen area where the folder structure is displayed shows the hierarchy to the current open folder only - folders on the same hierarchy level as e.g. folder A or sub-folder Aa are not displayed! Thus the vertical bar will be only shown if you open a folder with more content (bookmarks and sub-folders) than can be vertically displayed in the folder content area.

As I said, I have a lot of bookmarks and folders, nevertheless you can see in the screenshot just the folder structure and folder content of the folder opened - the rest of folder hierarchy is not visible (because on the same level as the top folder A). Therefore I think the original author called it flat bookmark as it's more than just removing the indentation, it's also displaying less thus making it less cluttered and more clearer.

**Comment by aaFn** [comment]
Hello @chr7 , ok I see, but this one is much more difficult to do, this is a deeper rewrite of the code.
I can easily vary the indentation and make it 0 or 1 pixel, however changing the nature of the display to only show the contents of a folder, and only showing its ancestors without their contents, is another matter.

Changing the issue title again then to reflect that.

At this stage, all I can do is vary the indentation. Let me know if useful or not. If not, then I'll defer this issue to some time when I have leisure to think about how to handle it.

**Comment by chr7** [comment]
I would like to give a smaller indentation a try. Will the number of pixels configurable? (as all folders on the same level will be displayed, 1 pixel might be too small to distinguish between content and folder on the same level.

**Comment by peter-kehl** [comment]
Hi @chr7, it seems that Flat Bookmarks History is discontinued. Did you find any replacement for it, please?

If not, then: this feature request would be so useful.

**Comment by chr7** [comment]
I'm surprised, I was not aware that this plugin might be not longer maintained. As I still use this plugin and it works so far, I haven't looked for any alternative.

**Comment by Gitoffthelawn** [comment]
@peter-kehl The `History in Threads` extension may be of help to you.

It is available on AMO here: https://addons.mozilla.org/firefox/addon/history-in-threads/
